components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/DataServiceDocLitWrappedSchemaGenerator.java

```diff
@@ -91,7 +91,7 @@ public static void populateServiceSchema(AxisService axisService) throws DataSer
             processRequest(cparams, resource);
         }
         /* process requestBox operation (Only have one element) */
-        if (dataservice.isBoxcarringEnabled()) {
+        if (dataservice.isBoxcarringEnabled() && dataservice.isDisableLegacyBoxcarringMode()) {

```

Comment:
Request box operation should be there always when boxcarring is enabled, regardless if the legacy operations are there or not. Basically, v3.5.0 functionality cannot break.